platform/search/src/services/GenericSearchProvider.js

```diff
@@ -171,17 +174,29 @@ define([
     GenericSearchProvider.prototype.index = function (id, model) {
         var provider = this;
 
-        this.worker.postMessage({
-            request: 'index',
-            model: model,
-            id: id
+        if (id !== 'ROOT') {
+            this.worker.postMessage({
+                request: 'index',
+                model: model,
+                id: id
+            });
+        }
+
+        var domainObject = objectUtils.toNewFormat(model, id);
+        var composition = _.find(this.openmct.composition.registry, function (p) {
+            return p.appliesTo(domainObject);
         });
 
-        if (Array.isArray(model.composition)) {
-            model.composition.forEach(function (idToIndex) {
-                provider.scheduleForIndexing(idToIndex);
-            });
+        if (!composition) {
+            return;
         }
+
+        composition.load(domainObject)
```

Comment:
The `_.find` through `composition.load` looks like it's replicating functionality of `composition.get(domainObject).load()`, which is public API. I'm unclear on why the private API is being used here instead; is it performance?